src/main/scala/scala/compat/java8/collectionImpl/Stepper.scala

```diff
@@ -0,0 +1,545 @@
+  }
+  
+  private class OfIntSpliterator(sp: Spliterator.OfInt)
+  extends IntStepper with java.util.function.IntConsumer {
+    private var cache: Int = 0
+    private var cached: Boolean = false
+    def accept(i: Int) { cache = i; cached = true }
+    
+    private def loadCache: Boolean = sp.tryAdvance(this)
+    private def useCache(c: java.util.function.IntConsumer): Boolean = {
+      if (cached) {
+        c.accept(cache)
+        cached = false
+        true
+      }
+      else false
+    }
+    
+    def characteristics = sp.characteristics
+    def estimateSize = {
+      val sz = sp.estimateSize
+      if (cached && sz < Long.MaxValue && sz >= 0) sz + 1
+      else sz
+    }
+    override def forEachRemaining(c: java.util.function.IntConsumer) {
+      useCache(c)
+      sp.forEachRemaining(c)
+    }
+    def hasNext = cached || loadCache
+    def nextInt = {
+      if (!hasNext) throwNSEE
+      val ans = cache
+      cached = false
+      ans
+    }
+    def substep(): IntStepper = {
+      val subSp = sp.trySplit()
+      if (subSp eq null) null
+      else {
+        val sub = new OfIntSpliterator(subSp)
+        if (cached) {
+          sub.cache = cache
+          sub.cached = true
+          cached = false
+        }
+        sub
+      }
+    }
+    override def tryAdvance(c: java.util.function.IntConsumer) = useCache(c) || sp.tryAdvance(c)
+  }
+  
+  private class OfLongSpliterator(sp: Spliterator.OfLong)
+  extends LongStepper with java.util.function.LongConsumer {
+    private var cache: Long = 0L
+    private var cached: Boolean = false
+    def accept(l: Long) { cache = l; cached = true }
+    
+    private def loadCache: Boolean = sp.tryAdvance(this)
+    private def useCache(c: java.util.function.LongConsumer): Boolean = {
+      if (cached) {
+        c.accept(cache)
+        cached = false
+        true
+      }
+      else false
+    }
+    
+    def characteristics = sp.characteristics
+    def estimateSize = {
+      val sz = sp.estimateSize
+      if (cached && sz < Long.MaxValue && sz >= 0) sz + 1
+      else sz
+    }
+    override def forEachRemaining(c: java.util.function.LongConsumer) {
+      useCache(c)
+      sp.forEachRemaining(c)
+    }
+    def hasNext = cached || loadCache
+    def nextLong = {
+      if (!hasNext) throwNSEE
+      val ans = cache
+      cached = false
+      ans
+    }
+    def substep(): LongStepper = {
+      val subSp = sp.trySplit()
+      if (subSp eq null) null
+      else {
+        val sub = new OfLongSpliterator(subSp)
+        if (cached) {
+          sub.cache = cache
+          sub.cached = true
+          cached = false
+        }
+        sub
+      }
+    }
+    override def tryAdvance(c: java.util.function.LongConsumer) = useCache(c) || sp.tryAdvance(c)
+  }
+  
+  /** Creates a `Stepper` over a generic `Spliterator`. */
+  def ofSpliterator[A](sp: Spliterator[A]): AnyStepper[A] = sp match {
+    case as: AnyStepper[A] => as
```

Comment:
What about specialized Steppers? Would it make sense to have a separate implementation that wraps them as AnyStepper instead of falling back to Spliterator?